components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -6574,4 +6574,276 @@ private void validateAndUpdateURITemplates(API api, int tenantId) throws APIMana
+    }
+
+    /**
+     * Retrieve the policies data added via the gateway policy mapping by mapping UUID.
+     *
+     * @param policyMappingUUID      Policy mapping UUID
+     * @param isWithPolicyDefinition This will decide whether to return policy definition or not, as policy definition
+     *                               is a bit bulky
+     * @return List of OperationPolicyData objects
+     * @throws APIManagementException
+     */
+    @Override
+    public List<OperationPolicyData> getGatewayPolicyDataListByPolicyId(String policyMappingUUID,
+            boolean isWithPolicyDefinition) throws APIManagementException {
+        return apiMgtDAO.getAllGatewayPoliciesDataForPolicyMappingUUID(policyMappingUUID, isWithPolicyDefinition);
+    }
+
+    /**
+     * Retrieve the policies added via the gateway policy mapping by mapping UUID.
+     *
+     * @param policyMappingUUID Policy mapping UUID
+     * @return List of OperationPolicy objects
+     * @throws APIManagementException
+     */
+    @Override
+    public List<OperationPolicy> getOperationPoliciesOfPolicyMapping(String policyMappingUUID)
+            throws APIManagementException {
+        return apiMgtDAO.getGatewayPoliciesOfPolicyMapping(policyMappingUUID);
+    }
+
+    /**
+     * Retrieve gateway policies mapping UUID attached to the gateway
+     *
+     * @param gatewayLabels List of gateway labels
+     * @param orgId         Organization ID
+     * @return Policy mapping UUID list
+     * @throws APIManagementException
+     */
+    @Override
+    public List<String> getAllPolicyMappingUUIDsByGatewayLabels(String[] gatewayLabels, String orgId)
+            throws APIManagementException {
+        return apiMgtDAO.getGatewayPolicyMappingByGatewayLabel(gatewayLabels, orgId);
+    }
+
+    /**
+     * Delete a gateway policy mapping.
+     *
+     * @param gatewayPolicyMappingId Gateway policy mapping UUID
+     * @throws APIManagementException if failed to delete comment for identifier
+     */
+    @Override
+    public void deleteGatewayPolicyMappingByPolicyMappingId(String gatewayPolicyMappingId, String tenantDomain)
+            throws APIManagementException {
+        apiMgtDAO.deleteGatewayPolicyMappingByPolicyId(gatewayPolicyMappingId, true);
+    }
+
+    /**
+     * Update globally added policies to the flows
+     *
+     * @param gatewayGlobalPolicyList List of Gateway Policy objects to be updated
+     * @param orgId                   Organization ID
+     * @param policyMappingId         Policy mapping UUID
+     * @return Policy Mapping ID
+     * @throws APIManagementException
+     */
+    @Override
+    public String updateGatewayGlobalPolicies(List<OperationPolicy> gatewayGlobalPolicyList, String description,
+            String name, String orgId, String policyMappingId) throws APIManagementException {
+        List<OperationPolicy> policyList = apiMgtDAO.getGatewayPoliciesOfPolicyMapping(policyMappingId);
+        if (policyList.isEmpty()) {
```

Comment:
can policyList be null any case?

Reply:
No, it will return an empty ArrayList even if no elements are retrieved from the database.